### PR TITLE
Modified go-netfilter-queue to support injecting new/modified packets

### PR DESCRIPTION
**Pull request by wkatsak**
I'd like to upstream this. Please let me know if you have any questions about my mods. As referenced in the commit message, these patches were based on the (outdated, non-working) patches at:
https://github.com/david415/go-netfilter-queue

I essentially used that concept, but cleaned up the code quite a bit and made it more understandable.

Thanks,
Bill Katsak

### CODE REVIEW

File: netfilter.h [hunk 1]
```diff
@@ -27,29 +27,36 @@
...
-    return nfq_set_verdict(qh, id, verdict, 0, NULL);
+    go_callback(id, buffer, ret, idx, &vc);
+
+    return nfq_set_verdict(qh, id, vc.verdict, vc.length, vc.data);
 }
 
 static inline struct nfq_q_handle* CreateQueue(struct nfq_handle *h, u_int16_t queue, u_int32_t idx)
 {
-  return nfq_create_queue(h, queue, &nf_callback, (void*)((uintptr_t)idx));
+    return nfq_create_queue(h, queue, &nf_callback, (void*)((uintptr_t)idx));
```
Comment (AkihiroSuda) [inline netfilter.h]: Could you fix indentation in a follow-up PR?
Thanks!

Comment (wkatsak) [reply netfilter.h]: Do you mean put it back to how it was?

Thanks,
Bill

Comment (AkihiroSuda) [reply netfilter.h]: uh, no problem, thanks

Comment (wkatsak) [reply netfilter.h]: Anything. To confirm, we are good as is, right?

Comment (AkihiroSuda) [reply netfilter.h]: yes, good as is

Comment (wkatsak) [reply netfilter.h]: Excellent. Thanks!

**Pull request merged**